app/Http/Controllers/GeneralLedgerAccountsController.php

```diff
@@ -34,52 +36,87 @@ public function create()
-    {
-        //
-    }
+    public function show(GeneralLedgerAccounts $chart_of_account)
+    {}
 
     /**
      * Show the form for editing the specified resource.
      *
-     * @param  \App\GeneralLedgerAccounts  $generalLedgerAccounts
+     * @param  \App\GeneralLedgerAccounts  $chart_of_account
      * @return \Illuminate\Http\Response
      */
-    public function edit(GeneralLedgerAccounts $generalLedgerAccounts)
+    public function edit(GeneralLedgerAccounts $chart_of_account)
     {
-        //
+        return view('settings.gla.edit', compact('chart_of_account'));
     }
 
     /**
      * Update the specified resource in storage.
      *
      * @param  \Illuminate\Http\Request  $request
-     * @param  \App\GeneralLedgerAccounts  $generalLedgerAccounts
+     * @param  \App\GeneralLedgerAccounts  $chart_of_account
      * @return \Illuminate\Http\Response
      */
-    public function update(Request $request, GeneralLedgerAccounts $generalLedgerAccounts)
+    public function update(Request $request, GeneralLedgerAccounts $chart_of_account)
     {
-        //
+        if (!$chart_of_account->update($request->all())) {
+            flash('Failed to update '.$chart_of_account->name)->error()->important();
+            return back();
+        }
+        flash('Successfully updated '.$chart_of_account->name)->success();
+        return redirect('/chart-of-accounts?type='.$chart_of_account->parent_identifier);
     }
 
     /**
      * Remove the specified resource from storage.
      *
-     * @param  \App\GeneralLedgerAccounts  $generalLedgerAccounts
+     * @param  \App\GeneralLedgerAccounts  $chart_of_account
      * @return \Illuminate\Http\Response
      */
-    public function destroy(GeneralLedgerAccounts $generalLedgerAccounts)
+    public function destroy(GeneralLedgerAccounts $chart_of_account)
+    {
+        if ($chart_of_account->balance > 0) {
+            flash('This account has a balance of '.$chart_of_account->balance.' therefore we cannot delete it.')->error()->important();
+            return back();
+        }
+        if (!$chart_of_account->delete()) {
+            flash('Failed to delete '. $chart_of_account->name)->error();
+            return back();
+        }
+        flash('Successfully deleted '.$chart_of_account->name)->success();
+        return redirect('/chart-of-accounts?type='.$chart_of_account->parent_identifier);
+    }
+
+    /**
+     * Show a form for creating a new gla under the 
+     * parent shown.
+     * @param GeneralLedgerAccounts $general_ledger_accounts [description]
+     * @param Request               $request                 [description]
+     */
+    public function addNewAccount(GeneralLedgerAccounts $general_ledger_accounts, Request $request)
+    {
+        return view('settings.gla.new', compact('general_ledger_accounts'));
+    }
+
+    public function saveNewAccount(GeneralLedgerAccounts $general_ledger_accounts, Request $request)
```

Comment:
found it. shoudn't u just merge this in the create and store respectively?

Reply:
custom url. need parent account data. store doesnt allow that, I would be violating laravels structure.